stellar_sdk/sep/stellar_web_authentication.py

```diff
@@ -54,6 +56,8 @@ def build_challenge_transaction(
     :param network_passphrase: The network to connect to for verifying and retrieving
         additional attributes from. (ex. 'Public Global Stellar Network ; September 2015')
     :param timeout: Challenge duration in seconds (default to 15 minutes).
+    :param client_domain: The domain of the client application requesting authentication
```

Comment:
Is it possible that `client_account_id` is the same address as `client_signing_key` in some special cases?

Reply:
`client_signing_key` is used to identify the client application, whereas `client_account_id` is used to identify the user for which the client application is acting on behalf of.

They would only be the same if the client application itself wanted to use a service (as a user) requiring SEP-10 authentication. I don't see a realistic use case for that.

Reply:
If the client application (`client_signing_key`) wanted to use a service as a user, it could just omit the `client_domain` operation because the challenge transaction would need to be signed by the secret key anyway.